src/DependencyInjection/FOSHttpCacheExtension.php

```diff
@@ -174,6 +174,33 @@ private function loadCacheControl(ContainerBuilder $container, array $config)
 
     private function parseRuleMatcher(ContainerBuilder $container, array $match)
     {
+        $requestMatcher = $this->parseRequestMatcher($container, $match);
+        $responseMatcher = $this->parseResponseMatcher($container, $match);
+
+        $signature = serialize([(string) $requestMatcher, (string) $responseMatcher]);
+        $id = $this->getAlias().'.cache_control.rule_matcher.'.md5($signature);
+
+        if (!$container->hasDefinition($id)) {
```

Comment:
the service ids of request + response matchers are based on the md5 of the rules in them. so if both are exactly the same as a previous rule, it could happen. 

hm, ok. while for response or request matcher alone that case makes sense, for a rule matcher it makes no sense as they are only used for cache_control and are the full definition of when a rule matches, hence a duplicate indicates a rule that is always hidden by a previous one.

should we instead throw an exception if the id already exists and complain that this rule is hidden by a previous one? could be helpful for the user to spot misconfiguration